contracts/StorageManager.sol

```diff
@@ -126,44 +123,31 @@ contract StorageManager {
         _setAvailableCapacity(offer, uint128(offer.availableCapacity.sub(decrease)));
     }
 
-     /**
-    @notice stops the Offer.
-    @dev when capacity is set to 0, no new Agreement can be created and no existing Agreement can be prolonged. All existing Agreement are still valid for the amount of periods still deposited.
-    */
+    /**
+   @notice stops the Offer.
+   @dev when capacity is set to 0, no new Agreement can be created and no existing Agreement can be prolonged. All existing Agreement are still valid for the amount of periods still deposited.
+   */
     function terminateOffer() public {
         Offer storage offer = offerRegistry[msg.sender];
         _setAvailableCapacity(offer, 0);
     }
 
     /**
+    >> FOR PROVIDER
     @notice set the billing plans for an Offer.
     @dev
     - setting the price to 0 means that a particular period is not offered, which can be used to remove a period from the offer.
     - make sure that any period * prices does not cause an overflow, as this can never be accepted (REF_MAX_PRICE) and hence is pointless.
-    - maximumDuration must be smaller or equal to the longest period (NOT verified by smart-contract).
     @param billingPeriods the offered periods. Length must be equal to billingPrices.
     @param billingPrices the prices for the offered periods. Each entry at index corresponds to the same index at periods. 0 means that the particular period is not offered.
     */
     function setBillingPlans(uint64[] memory billingPeriods, uint64[] memory billingPrices) public {
```

Comment:
missing the requires
        require(billingPeriods.length > 0, "StorageManager: Offer needs some billing plans");
        require(billingPeriods.length == billingPrices.length, "StorageManager: Billing plans array length has to equal to billing prices");

If I execute setBillingPlans before setOffer, it is not a problem because it is possible at terminateOffer. Is this correct?

Reply:
Well, it would leave the Offer in a weird state, because currently it is tracked if Offer exists using the `availableCapacity` so it would have set billing plans, but would be considered as it does not exist.

I fixed this with the `existingOffer` modifier. Thanks for the requires.